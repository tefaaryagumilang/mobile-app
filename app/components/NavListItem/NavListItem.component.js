import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View, Text} from 'react-native';
import SimasIcon from '../../assets/fonts/SimasIcon';
import Touchable from '../Touchable.component';
import styles from './NavListItem.styles';
import {Toast} from '../../utils/RNHelpers.util.js';
import {language} from '../../config/language';


export default class NavListItem extends Component {
  static propTypes = {
    label: PropTypes.string,
    subtitle: PropTypes.string,
    onPress: PropTypes.func,
    featureIconName: PropTypes.string,
    footer: PropTypes.string,
    disabled: PropTypes.bool,
    style: PropTypes.object,
    iconSize: PropTypes.number,
    theme: PropTypes.oneOf(['primary', 'plain', 'depositButtonStyle']),
    status: PropTypes.string,
    dtActionName: PropTypes.string
  }
  static defaultProps = {
    subtitle: null
  }

  toastComingSoon = () => {
    const {status = ''} = this.props;
    Toast.show(status === 'disable' ? language.DASHBOARD__CREDIT_CARD_DISABLE_CA : language.DASHBOARD__CREDIT_CARD_SOON);
  }

  render () {
    const {featureIconName, iconSize = 30, label, subtitle, onPress, disabled = false, style, theme = 'primary', footer = '', status = '', dtActionName} = this.props;
    let navItem = null;
    const comingsoon = language.CGV_COMING_SOON;
    const isBillerTypeFive = dtActionName === 'Credit Card Bill Pay';
    if (theme === 'primary') {
      navItem = (
        <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card Primary' : dtActionName} onPress={!disabled ? onPress : null} underlayColor={theme.primary} onShowUnderlay={this.showUnderlay} onHideUnderlay={this.onHideUnderlay}>
          <View style={[styles.container, style, disabled ? styles.disabled : {}]}>
            <View style={styles.paddingContent}>
              <View style={styles.buttonContainer}>
                <View style={styles.featureIconContainer}>
                  {featureIconName && <SimasIcon name={featureIconName} size={iconSize} style={styles.featureIcon}/>}
                </View>
                <View style={styles.labelContainer}>
                  <Text style={styles.title}>{label}</Text>
                  {subtitle ? <Text style={styles.subtitle}>{subtitle}</Text> : null}
                  {footer ? <Text style={styles.footer}>{footer}</Text> : null}
                </View>
                <View style={styles.navIconContainer}>
                  <SimasIcon name={'arrow'} size={10} style={styles.navIcon}/>
                </View>
              </View>
            </View>
          </View>
        </Touchable>);
      return navItem;
    }
    if (theme === 'plain') {
      navItem = (<View style={styles.optionItemContainer}>
        <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card Plain' : dtActionName} style={styles.optionItem} onPress={onPress}>
          <Text style={styles.optionText}>{label}</Text>
          <SimasIcon name='arrow' style={styles.navIcon} />
        </Touchable>

      </View>);
      return navItem;
    }

    if (theme === 'none') {
      navItem = (<View style={styles.optionItemContainer}>
        <View style={styles.optionItem} onPress={onPress}>
          <Text style={styles.optionText}>{label}</Text>
        </View>
      </View>);
      return navItem;
    }

    if (theme === 'VCC') {
      navItem = (<View>
        <View style={styles.optionBox}>
          {status === 'soon' ?
            <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card VCC Status ' + status : dtActionName} onPress={this.toastComingSoon} style={styles.optionItem}>
              <View style={styles.boxstatus}>
                <Text style={styles.optionTextV}>{label}</Text>
                <View style={status === 'new' ? styles.optionRed : status === 'soon' ? styles.optionKet : null}>
                  <Text style={styles.optionTextKet}>{status === 'new' ? status : status === 'soon' ? comingsoon : null}</Text>
                </View>
              </View>
              <SimasIcon name='arrow' style={styles.navIcon} />
            </Touchable>
            : status === 'disable' ?
              <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card VCC Status ' + status : dtActionName} onPress={this.toastComingSoon} style={styles.optionItem}>
                <View style={styles.boxstatus}>
                  <Text style={styles.optionTextV}>{label}</Text>
                  <View style={status === 'new' ? styles.optionRed : status === 'soon' ? styles.optionKet : null}>
                    <Text style={styles.optionTextKet}>{status === 'new' ? status : status === 'soon' ? comingsoon : null}</Text>
                  </View>
                </View>
                <SimasIcon name='arrow' style={styles.navIcon} />
              </Touchable>
              :
              <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card VCC Status ' + status : dtActionName} style={styles.optionItem} onPress={onPress}>
                <View style={styles.boxstatus}>
                  <Text style={styles.optionTextV}>{label}</Text>
                  <View style={status === 'new' ? styles.optionRed : status === 'soon' ? styles.optionKet : null}>
                    <Text style={styles.optionTextKet}>{status === 'new' ? status : status === 'soon' ? comingsoon : null}</Text>
                  </View>
                </View>
                <SimasIcon name='arrow' style={styles.navIcon} />
              </Touchable> }
        </View>

        <View style={styles.summaryArea}/>
      </View>);
      return navItem;
    }

    if (theme === 'VCCN') {
      navItem = (<View>
        <View style={styles.optionBox}>
          {status === 'soon' ?
            <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card VCCN Status ' + status : dtActionName} onPress={this.toastComingSoon} style={styles.optionItem}>
              <View style={styles.boxstatus}>
                <Text style={styles.optionTextV}>{label}</Text>
                <View style={status === 'new' ? styles.optionRed : status === 'soon' ? styles.optionKet : null}>
                  <Text style={styles.optionTextKet}>{status === 'new' ? status : status === 'soon' ? comingsoon : null}</Text>
                </View>
              </View>
              <SimasIcon name='arrow' style={styles.navIcon} />
            </Touchable>
            :
            <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card VCCN Status ' + status : dtActionName} style={styles.optionItem} onPress={onPress}>
              <View style={styles.boxstatus}>
                <Text style={styles.optionTextV}>{label}</Text>
                <View style={status === 'new' ? styles.optionRed : status === 'soon' ? styles.optionKet : null}>
                  <Text style={styles.optionTextKet}>{status === 'new' ? status : status === 'soon' ? comingsoon : null}</Text>
                </View>
              </View>
              <SimasIcon name='arrow' style={styles.navIcon} />
            </Touchable>}
        </View>

      </View>);
      return navItem;
    }

    if (theme === 'depositButtonStyle') {
      navItem = (
        <Touchable dtActionName={isBillerTypeFive ? dtActionName + ' - Credit Card Deposit Status ' + status : dtActionName} onPress={!disabled ? onPress : null} underlayColor={theme.primary} onShowUnderlay={this.showUnderlay} onHideUnderlay={this.onHideUnderlay}>
          <View style={[styles.containerDeposit, style, disabled ? styles.disabled : {}]}>
            <View style={styles.buttonContainer}>
              <View style={styles.featureIconContainer}>
                {featureIconName && <SimasIcon name={featureIconName} size={iconSize} style={styles.featureIcon}/>}
              </View>
              <View style={styles.labelContainer}>
                <Text style={styles.title}>{label}</Text>
                {subtitle ? <Text style={styles.subtitle}>{subtitle}</Text> : null}
                {footer ? <Text style={styles.footer}>{footer}</Text> : null}
              </View>
              <View style={styles.navIconContainer}>
                <SimasIcon name={'arrow'} size={10} style={styles.navIcon}/>
              </View>
            </View>
          </View>
        </Touchable>);
      return navItem;
    }
    return null;
  }
}
