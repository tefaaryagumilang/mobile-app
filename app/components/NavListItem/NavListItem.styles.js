import {theme} from '../../styles/core.styles';
import * as styles from '../../styles/common.styles';

export default {
  container: [
    styles.borderGreyStyle,
    {
      borderBottomWidth: theme.separatorSize,
      borderBottomColor: theme.separatorColor,
      paddingVertical: 30
    }
  ],
  containerDeposit: [
    styles.borderGreyStyle,
    {
      borderBottomWidth: 1,
      paddingVertical: 10
    }
  ],
  buttonContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  labelContainer: {
    flex: 1,
    paddingLeft: 20,
    paddingRight: 10,
    alignItems: 'flex-start',
    justifyContent: 'flex-start'
  },
  title: [
    styles.fontSizeNormalStyle,
    styles.textBrandStyle
  ],
  subtitle: [
    styles.fontSizeSmallStyle,
    styles.textBrandStyle, {
      paddingTop: 5
    }
  ],
  footer: [
    styles.fontSizeSmallStyle, {
      color: theme.lightGrey,
      paddingTop: 5
    }],
  featureIconContainer: {
    flex: 0,
    width: 40,
    height: 30,
    justifyContent: 'center',
  },
  featureIcon: {
    color: theme.brand,
  },
  navIconContainer: {
    flex: 0,
    justifyContent: 'center',
    alignItems: 'flex-end',
    width: 30,
    height: 30,
  },
  navIcon: {
    color: theme.black,
  },
  disabled: {
    opacity: 0.5
  },
  optionItemContainer: {
    borderBottomWidth: 1,
    borderBottomColor: theme.borderGrey,
  },
  optionItem: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 15,
    alignItems: 'center',
  },
  optionTextV: [styles.fontSizeMediumStyle, {
    color: theme.darkBlue
  }],

  optionText: [styles.fontSizeMediumStyle],
  paddingContent: {
    paddingHorizontal: 20,
  },
  summaryArea: {
    borderBottomWidth: 0.7,
    borderColor: theme.grey,
    marginTop: 10,
    marginBottom: 10
  },
  optionBox: {
    paddingHorizontal: 20
  },
  optionTextKet: [styles.fontSizeXSStyle, {
    color: theme.white
  }],
  optionRed: {
    backgroundColor: theme.brand,
    paddingHorizontal: 5,
    borderRadius: 8,
    marginLeft: 5,
    alignItems: 'center',
    marginBottom: 8
  },
  optionKet: {
    backgroundColor: theme.textLightGrey,
    paddingHorizontal: 5,
    borderRadius: 8,
    marginLeft: 5,
    alignItems: 'center',
    marginBottom: 8
  },
  boxstatus: {
    flexDirection: 'row'
  },
};
