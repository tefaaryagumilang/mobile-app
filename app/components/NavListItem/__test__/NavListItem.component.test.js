import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import {reduxForm} from 'redux-form';
import NavListItem from '../NavListItem.component';

const spy = jest.fn();
const store = createStore(() => ({}));
const DecoratedForm = reduxForm({form: 'testForm'})(NavListItem);

describe('OnboardingJourney Component: EasyPinView', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <DecoratedForm handleSubmit={spy}/>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
