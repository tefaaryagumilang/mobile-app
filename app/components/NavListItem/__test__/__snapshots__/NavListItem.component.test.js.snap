// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingJourney Component: EasyPinView renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Array [
          Object {
            "borderColor": "rgba(0,0,0, 0.09)",
          },
          Object {
            "borderBottomColor": "#ECECEC",
            "borderBottomWidth": 8,
            "paddingVertical": 30,
          },
        ],
        undefined,
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "height": 30,
              "justifyContent": "center",
              "width": 40,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "justifyContent": "flex-start",
              "paddingLeft": 20,
              "paddingRight": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 14,
                },
                Object {
                  "color": "#FA3F70",
                },
              ]
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 0,
              "height": 30,
              "justifyContent": "center",
              "width": 30,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 10,
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontFamily": "simas",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î¤†
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
